### ISSUE TITLE
Contact managers not getting updated when adding links

### ISSUE THREAD

**Issue by mpowelson** [ros-industrial-consortium/tesseract#582]
I don't have any shareable code to replicate this, but it seems that when adding links with replace turned on (may not be relevant) at least some times the contact managers don't get correctly updated. I'm not 100% sure this really the case, but something seems off.

**Comment by Levi-Armstrong** [comment]
I have found the issue. It is an interesting issue. Bullet does some optimizations to prevent updating things so if you change the active links and not change the contact distance things in the broadphase do not update. In the unit test you created if you change the contact distance to something small like 0.0000000001 it works. Though if you increase this but move set active links after setting the contact distance it no longer works. I did some digging in bullet and found found a solution but need to create a unit test in tesseract collision. My changes are current on top the most recent PR so I create a new PR once it is merged with only the change to tesseract_collision.
